modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -415,12 +429,19 @@ protected Result beginEncode(HttpResponse httpResponse, String acceptEncoding) t
             if (ResourceLeakDetector.isEnabled()) {
                 ch.pipeline().addLast(new Netty4LeakDetectionHandler());
             }
+            // See https://github.com/netty/netty/issues/15053: the combination of FlowControlHandler and HttpContentDecompressor above
+            // can emit multiple chunks per read, but HttpBody.Stream requires chunks to arrive one-at-a-time so until that issue is
+            // resolved we must add another flow controller here:
+            ch.pipeline().addLast(new FlowControlHandler());
```

Comment:
This is the bit that was missing from https://github.com/elastic/elasticsearch/pull/126441 - it's covered by tests added in  #127237 and #127111.